src/components/TodoForm.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import { useTodoContext } from '../context/TodoContext';
+import { addTodo } from '../api/todos';
+import { TodoError } from '../types/Error';
+import { USER_ID } from '../utils/constants';
+
+export const TodoForm: React.FC = () => {
+  const {
+    setTodos,
+    setError,
+  } = useTodoContext();
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isCreating, setIsCreating] = useState(false);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setError(TodoError.INVALID_INPUT);
+
+      return;
+    }
+
+    setError(null);
+    setIsCreating(true);
+
+    const newTodo = {
+      userId: USER_ID,
+      title: todoTitle,
+      completed: false,
+    };
+
+    try {
+      const todo = await addTodo(newTodo);
```

Comment:
In theory, the form component shouldn't do any work other than preparing the data. This can be done in some parent component to share responsibility.

For example, a doctor prescribes medicine, but does not sell it. It is up to the patient to decide where, when and how to purchase them. In the same way, the form collects, validates and prepares data like a recipe and passes it on. And then it depends on what is needed: either upload data to the server or do something else.

Please think of this as the Principles of Single Responsibility